static/js/components/auth/AuthEmailForm.js

```diff
@@ -2,49 +2,114 @@
 
 import type { EmailForm } from "../../flow/authTypes"
 import type { FormProps } from "../../flow/formTypes"
 
-const AuthEmailForm = ({
-  form,
-  validation,
-  onSubmit,
-  onUpdate,
-  onRecaptcha,
-  processing
-}: FormProps<EmailForm>) => {
-  return (
-    <form onSubmit={onSubmit} className="form">
-      <div className="emailfield row">
-        <input
-          type="email"
-          name="email"
-          value={form.email}
-          onChange={onUpdate}
-          placeholder="Email"
-        />
-        {validationMessage(validation.email)}
-      </div>
-      {SETTINGS.recaptchaKey && onRecaptcha ? (
-        <div className="recaptcha row">
-          <ReCAPTCHA sitekey={SETTINGS.recaptchaKey} onChange={onRecaptcha} />
-          {validationMessage(validation.recaptcha)}
+type Props = FormProps<EmailForm>
+type State = {
+  recaptchaScale: number
+}
+
+// These needs to stay in sync with ReCAPTCHA's runtime width
+const RECAPTCHA_WIDTH = 304
+const RECAPTCHA_HEIGHT = 78
+
+const StyledReCAPTCHA = styled(ReCAPTCHA)`
+  transform: scale(${props => props.scale.toFixed(3)});
+  transform-origin: 0 0;
+  height: ${props => (props.scale * RECAPTCHA_HEIGHT).toFixed(0)}px;
+`
+
+class AuthEmailForm extends React.Component<Props, State> {
+  recaptcha: null
+
+  constructor(props: Props) {
+    super(props)
+    this.recaptcha = null
+    // only rescale up to 4x a second to salvage some performance
+    this.scaleRecaptcha = debounce(this.scaleRecaptcha.bind(this), 250)
+    this.state = {
+      recaptchaScale: 1.0
+    }
+  }
+
+  componentDidMount() {
+    window.addEventListener("resize", this.scaleRecaptcha)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("resize", this.scaleRecaptcha)
+  }
+
+  scaleRecaptcha = () => {
+    if (this.recaptcha) {
+      const {
+        captcha: { clientWidth }
+      } = this.recaptcha
+      // compute this as a fractional scale of the scale that ReCAPTCHAwants to render at if our container is smaller
+      const recaptchaScale =
+        clientWidth < RECAPTCHA_WIDTH ? clientWidth / RECAPTCHA_WIDTH : 1.0
+
+      this.setState({ recaptchaScale })
+    }
+  }
+
+  setRecaptcha = (recaptcha: any) => {
+    this.recaptcha = recaptcha
+    this.scaleRecaptcha()
+  }
+
+  render() {
+    const {
+      form,
+      validation,
+      onSubmit,
+      onUpdate,
+      onRecaptcha,
+      processing
+    } = this.props
+    const { recaptchaScale } = this.state
+
+    return (
+      <form onSubmit={onSubmit} className="form">
+        <div className="emailfield row">
+          <input
+            type="email"
+            name="email"
+            value={form.email}
+            onChange={onUpdate}
+            placeholder="Email"
+          />
+          {validationMessage(validation.email)}
+        </div>
+        {SETTINGS.recaptchaKey && onRecaptcha ? (
+          <div className="recaptcha row">
+            <StyledReCAPTCHA
+              ref={this.setRecaptcha}
```

Comment:
should we use the newer `ref` API here, with `React.createRef` and so on? I think then the `scaleCaptcha` method might have to be called in `componentDidMount` or something, not sure if the specific use-case here prevents us from using that API

Reply:
So the reason I didn't do that is because I needed some hook to call when the component first instantiates and I have a realistic idea of the size of the container, when the ref is created seemed like the best in practice unless you have a better suggestion.

Reply:
hmm, yeah that probably makes sense. I think there are a few guarantees that refs will be populated at certain points in the component life cycle, but I think if we let the DOM elements be rendered without immediately grabbing the ref we could potentially run into a situation where the element flashes on the screen at a wider width for a few frames, which we probably want to avoid. So yeah, I think it's fine to go with this.